### PR TITLE
Implemented Events

### PR DESCRIPTION
**Pull request by Mekuseo**
- Create a new directory called logic inside your project
- Copy the files calculate.js and operate.js into your logic/ directory
- Install big.js using NPM
- Analyze the files calculate.js and operate.js, be sure you understand what they do
Import the files in your Calculator.js component
Implement click event handlers using the math logic from calculate.js and operate.js in your React component
- Test your app in the browser. The Calculator should be able to perform math operations.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,111 +1,134 @@
...
+import calculate from './logic/Calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
-    super();
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      // operation: null,
```
Comment (GedeonTS) [inline src/components/Calculator.js]: - [x] Kindly remove all the commented out lines of code here to make you code cleaner and professional :recycle:

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,111 +1,134 @@
...
           </button>
           <button
             type="submit"
             className="calculator__key calculator__key--operator"
             value="+/-"
+            onClick={this.handleClick}
           >
             +/-
           </button>
           <button
```
Comment (GedeonTS) [inline src/components/Calculator.js]: - [ ] As per the proct requirements [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md#:~:text=Note%3A%20Practice%20what%20you%20have%20learned%20until%20now%3A%20use%20useState%20and%20props.) kindly make sure to use props in your project by using child components.

__Hints: for this you might have to create a component button that you will use everywhere you need button, giving it the handleclick function and the right value as props __

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @Mekuseo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- No linter errors :+1: 
- Good use of Gitflow :+1: 
- The Calculator is working fine :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL]  Kindly make sure to write more descriptive commit messages, using the imperative form and describing clearly your changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (Mekuseo) [reply src/components/Calculator.js]: Done

**Review by sumairq — APPROVED** [review]
### Hi @Mekuseo , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**